src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.CreateToken.cs

```diff
@@ -754,8 +762,13 @@ int sizeOfEncodedHeaderAndPayloadAsciiBytes
                         audienceChecked = true;
                         if (audienceSet)
                         {
+                            string descriptorMemberName = null;
+                            if (!tokenDescriptor.Audiences.IsNullOrEmpty())
```

Comment:
do not allocate the string descriptorMemberName unless you are going to log something.

Reply:
moved EventLogLevel check up a level so the allocation only happens if we are logging it.